### PR TITLE
Switch from TravisCI to Github Actions

### PR DESCRIPTION
**Pull request by jreese**
Removes the TravisCI config and replaces it with a Github Actions config that should replicate the tests on 3.6-3.8, but also run them on Linux, macOS, and Windows.

### CODE REVIEW

**Comment by jreese** [comment]
Runs CI on pull requests, new tags, and pushes to master branch. Separated the lint and coverage steps from tests so that coverage could fail without failing the whole CI job. Got a "clean" run my own repo: https://github.com/jreese/modernize/actions/runs/237080751

Caveats are that coveralls needs the API token to either be a) added as a secret named COVERALLS_REPO_TOKEN to the Github repo, or b) added to build.yml or .coveralls.yml directly. The problem with A is that coveralls will not work for pull requests from unprivileged users, but B exposes the token as part of the repo, so could potentially be "abused".

File: .github/workflows/build.yml [hunk 1]
```diff
@@ -0,0 +1,43 @@
...
+          python-version: ${{ matrix.python-version }}
+      - name: Set Up Node ${{ matrix.node-version }}
+        uses: actions/setup-node@v1
+        with:
+          node-version: ${{ matrix.node-version }}
+      - name: Install
+        run: |
+          pip install tox
+      - name: Test
+        run: tox -e py${{ matrix.python-version[0] }}${{ matrix.pythos-version[2] }}
```
Comment (graingert) [inline .github/workflows/build.yml:36]: `tox -e py,lint,coveralls` should do it

**Comment by graingert** [comment]
> Runs CI on pull requests, new tags, and pushes to master branch. Separated the lint and coverage steps from tests so that coverage could fail without failing the whole CI job. Got a "clean" run my own repo: [jreese/modernize/actions/runs/237080751](https://github.com/jreese/modernize/actions/runs/237080751)
You should get cleaner steps if we can get tox to output ` ::group ` or `##[group]`, also coveralls is a required status of all builds anyway - so I think we shouldn't ignore it in the action either

> Caveats are that coveralls needs the API token to either be a) added as a secret named COVERALLS_REPO_TOKEN to the Github repo, or b) added to build.yml or .coveralls.yml directly. The problem with A is that coveralls will not work for pull requests from unprivileged users, but B exposes the token as part of the repo, so could potentially be "abused".

I think you can just add this passenv and have it work in PRs: https://coveralls-python.readthedocs.io/en/latest/usage/tox.html#github-actions

**Comment by graingert** [comment]
@jreese there's now a "hello-world" action in the default branch

File: .github/workflows/build.yml [hunk 2]
```diff
@@ -0,0 +1,43 @@
...
+      - name: Lint
+        run: tox -e lint
+      - name: Coveralls
+        env:
+          COVERALLS_REPO_TOKEN: ${{ secrets.COVERALLS_REPO_TOKEN }}
+        run: tox -e coveralls
+        continue-on-error: true
```
Comment (graingert) [inline .github/workflows/build.yml:43]:
```suggestion
      - name: Coveralls
        env:
          GITHUB_TOKEN: ${{ secrets.GITHUB_TOKEN }}
        run: tox -e coveralls
```

**Pull request closed without merging**